sigmastate/src/test/scala/sigmastate/serialization/ContractTemplateSerializerSpecification.scala

```diff
@@ -0,0 +1,71 @@
+package sigmastate.serialization
+
+import org.ergoplatform.SigmaConstants.ScriptCostLimit
+import org.scalatest.Assertion
+import sigmastate.Values.ErgoTree.{ConstantSegregationHeader, EmptyConstants}
+import sigmastate.Values.{BigIntConstant, ByteConstant, ContractTemplate, ErgoTree, IntConstant, Parameter, ShortConstant, SigmaPropValue}
+import sigmastate.eval.{CBigInt, IRContext}
+import sigmastate.{CrossVersionProps, DataValueComparer, EQ, JitCost, Plus, SType, Values}
+import sigmastate.helpers.SigmaTestingCommons
+import sigmastate.interpreter.{CostAccumulator, ErgoTreeEvaluator}
+import sigmastate.interpreter.ErgoTreeEvaluator.DefaultProfiler
+import sigmastate.serialization.ErgoTreeSerializer.DefaultSerializer
+
+import java.math.BigInteger
+
+class ContractTemplateSerializerSpecification extends SerializationSpecification
+  with SigmaTestingCommons with CrossVersionProps {
+
+  implicit lazy val IR: TestingIRContext = new TestingIRContext {
+    beginPass(noConstPropagationPass)
+  }
+
+  def roundtrip(template: ContractTemplate): Assertion = {
+    val bytes = ContractTemplateSerializer.DefaultSerializer.serializeContractTemplate(template)
+    val res2 = ContractTemplateSerializer.DefaultSerializer.deserializeContractTemplate(bytes)
+    res2 shouldEqual template
+  }
+
+  private def extractConstants(prop: SigmaPropValue)(implicit IR: IRContext): (IndexedSeq[Values.Constant[SType]], SigmaPropValue) = {
+    val env = Map[String, Any]()
+    val res = compiler.compileTyped(env, prop)
+    checkCompilerResult(res)
+    val calcF = res.compiledGraph
+    val constantsStore = new ConstantStore()
+    val outExpr = IR.buildTree(calcF, Some(constantsStore))
+    val constants = constantsStore.getAll
+    (constants, outExpr)
+  }
```

Comment:
You should never need the compiler for template operations. Compiler is only used for text => ErgoTree transformation and we don't have to do with `text` in templates.

Reply:
Yeah, I just wanted to use the compiler to extract the constants from the Tree. Not needed though and I've updated the code.